File: frontend/src/api/agents.ts
export interface AgentRun {
  id: string
  agent_name: string
  context: string
  status: string
  duration_ms: number
  error_message?: string
  created_at: string
  completed_at?: string
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  input_summary?: any
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  output_summary?: any
}

export interface AgentRunListResponse {
  items: AgentRun[]
  total: number
  page: number
  limit: number
}

export interface AgentSummary {
  total_runs: number
  runs_by_agent: Record<string, number>
  success_rate: number
  recent_failures: Array<{
    id: string
    agent_name: string
    error_message: string
    created_at: string
  }>
}

export async function fetchAgentRuns(
  page: number = 1, 
  limit: number = 20, 
  agentName?: string, 
  status?: string
): Promise<AgentRunListResponse> {
  const params = new URLSearchParams({
    page: page.toString(),
    limit: limit.toString()
  })
  
  if (agentName) params.append('agent_name', agentName)
  if (status) params.append('status', status)
  
  const response = await fetch(`/api/agents/runs?${params.toString()}`)
  if (!response.ok) {
    throw new Error('Failed to fetch agent runs')
  }
  return response.json()
}

export async function fetchAgentRunDetails(runId: string): Promise<AgentRun> {
  const response = await fetch(`/api/agents/runs/${runId}`)
  if (!response.ok) {
    throw new Error('Failed to fetch agent run details')
  }
  return response.json()
}

export async function fetchAgentSummary(): Promise<AgentSummary> {
  const response = await fetch('/api/agents/summary')
  if (!response.ok) {
    throw new Error('Failed to fetch agent summary')
  }
  return response.json()
}

File: backend/api/routes/agents.py
"""
Agents API routes
"""

from typing import Dict, List, Optional

from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy import func
from sqlalchemy.orm import Session

from backend.api.database import get_db
from backend.api.models import AgentRunLog, User

router = APIRouter(prefix="/agents", tags=["agents"])


@router.get("/runs")
async def get_agent_runs(
    page: int = 1,
    limit: int = 20,
    agent_name: Optional[str] = None,
    status: Optional[str] = None,
    db: Session = Depends(get_db),
):
    """List agent runs with pagination and filtering."""
    # TODO: Get user from session/auth token
    user = db.query(User).first()
    if not user:
        return {"items": [], "total": 0, "page": page, "limit": limit}

    query = db.query(AgentRunLog).filter(AgentRunLog.user_id == user.id)

    if agent_name:
        query = query.filter(AgentRunLog.agent_name == agent_name)

    if status:
        query = query.filter(AgentRunLog.status == status)

    total = query.count()

    runs = query.order_by(AgentRunLog.created_at.desc()).offset((page - 1) * limit).limit(limit).all()

    return {
        "items": [
            {
                "id": str(run.id),
                "agent_name": run.agent_name,
                "context": run.context,
                "status": run.status,
                "duration_ms": run.duration_ms,
                "error_message": run.error_message,
                "created_at": run.created_at.isoformat() if run.created_at else None,
                "completed_at": run.completed_at.isoformat() if run.completed_at else None,
                # Include brief summary for list view
                "input_summary": run.input_summary,
                "output_summary": run.output_summary,
            }
            for run in runs
        ],
        "total": total,
        "page": page,
        "limit": limit,
    }


@router.get("/runs/{run_id}")
async def get_agent_run_details(run_id: str, db: Session = Depends(get_db)):
    """Get detailed agent run log."""
    # TODO: Get user from session/auth token
    user = db.query(User).first()
    if not user:
        raise HTTPException(status_code=401, detail="User not found")

    run = db.query(AgentRunLog).filter(AgentRunLog.id == run_id, AgentRunLog.user_id == user.id).first()

    if not run:
        raise HTTPException(status_code=404, detail="Agent run not found")

    return {
        "id": str(run.id),
        "agent_name": run.agent_name,
        "context": run.context,
        "status": run.status,
        "duration_ms": run.duration_ms,
        "error_message": run.error_message,
        "created_at": run.created_at.isoformat() if run.created_at else None,
        "completed_at": run.completed_at.isoformat() if run.completed_at else None,
        "input_summary": run.input_summary,
        "output_summary": run.output_summary,
    }


@router.get("/summary")
async def get_agent_summary(db: Session = Depends(get_db)):
    """Get summary statistics for agents."""
    # TODO: Get user from session/auth token
    user = db.query(User).first()
    if not user:
        return {}

    # Total runs
    total_runs = db.query(AgentRunLog).filter(AgentRunLog.user_id == user.id).count()

    # Runs by agent
    runs_by_agent = (
        db.query(AgentRunLog.agent_name, func.count(AgentRunLog.id))
        .filter(AgentRunLog.user_id == user.id)
        .group_by(AgentRunLog.agent_name)
        .all()
    )

    # Success rate
    success_count = (
        db.query(AgentRunLog).filter(AgentRunLog.user_id == user.id, AgentRunLog.status == "success").count()
    )

    success_rate = (success_count / total_runs * 100) if total_runs > 0 else 0

    # Recent failures
    recent_failures = (
        db.query(AgentRunLog)
        .filter(AgentRunLog.user_id == user.id, AgentRunLog.status == "error")
        .order_by(AgentRunLog.created_at.desc())
        .limit(5)
        .all()
    )

    return {
        "total_runs": total_runs,
        "runs_by_agent": {name: count for name, count in runs_by_agent},
        "success_rate": round(success_rate, 2),
        "recent_failures": [
            {
                "id": str(run.id),
                "agent_name": run.agent_name,
                "error_message": run.error_message,
                "created_at": run.created_at.isoformat() if run.created_at else None,
            }
            for run in recent_failures
        ],
    }
